docs/ansible/coding-conventions.md

```diff
@@ -0,0 +1,1178 @@
+          repo: 'https://{{gittoken}}@github.com/AKSarav/SampleNodeApp.git'
+          dest: "{{ destdir }}"
+     - name: Change the ownership of the directory
+        become: yes
+        file:
+          path: "{{destdir}}"
+          owner: "vagrant"
+        register: chgrpout
+     - name: Install Dependencies with NPM install command
+        shell:
+          "npm install"
+        args:
+          chdir: "{{ destdir }}"
+        register: npminstlout
+     - name: Debug npm install command
+        debug: msg='{{npminstlout.stdout_lines}}'
+     - name: Start the App
+        async: 10
+        poll: 0
+        shell:
+          "(node index.js > nodesrv.log 2>&1 &)"
+        args:
+          chdir: "{{ destdir }}"
+        register: appstart
+     - name: Validating the port is open
+        tags: nodevalidate
+        wait_for:
+          host: "localhost"
+          port: 5000
+          delay: 10
+          timeout: 30
+          state: started
+          msg: "NodeJS server is not running"
+  post_tasks:
+     - name: notify Slack that the servers have been updated
+        tags: slack
+        community.general.slack:
+          token: T026******PF/B02U*****N/WOa7r**********Ao0jnWn
+          msg: |
+             ### StatusUpdate ###
+             – ------------------------------------
+             ``
+             `Server`: {{ansible_host}}
+             `Status`: NodeJS Sample Application installed successfully
+             – ------------------------------------
+          channel: '#ansible'
+          color: good
+          username: 'Ansible on {{ inventory_hostname }}'
+          link_names: 0
+          parse: 'none'
+        delegate_to: localhost
+```
+
+>**🔎 Why do this?**
+>
+>This creates a pretty and tidy code which is easy to read, both for dyslectic and non dyslectic people. There is no excuse not to do this.
+
+
+
+## Order in playbook
+
+Playbook definitions should follow this order.
+
+* `name`
+* `hosts`
+* `remote_user`
+* `become`
+* `vars`
+* `pre_tasks`
+* `roles`
+* `tasks`
+
+✅ ***Good***
+```yaml
+---
+- name: update root authorized_keys for all machines
+  hosts:
+    - all
+  become: true
+
+  vars:
+    root_keys_users:
+      - user: user1
+        key: |
+          ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHkCVF05JvfkrfOOESivOxV4N8+A/EMEkF7/nCQMRoQg
+        enabled: true
+
+  pre_tasks:
+    - name: add custom root users
+      set_fact:
+        root_keys_users: "{{ root_keys_users | union( root_keys_users_custom|default([]) ) }}"
+
+  roles:
+    - role: root-keys
+
+  tasks:
+    - name: print out debug message
+      ansible.builtin.debug:
+        msg: "fee foo faa"
+```
+
+>**🔎 Why do this?**
+>
+>A common order makes playbooks consistent and easier to read for your dear colleagues. Think of them when you write.
+
+## Order in task declaration
+
+A task should be declared in this order.
+
+* `name`
+* module and the arguments for the module
+* arguments for the task in correct order
```

Comment:
add red bulb and bold TODO in brackets to define should this be applied and update this item with more details or delete it

Reply:
I already made all you requested forr this part: changed 'alphabetical' to 'correct' and gave example.
Don't think that TODO required here.